docs/docsite/rst/contributor_guide.rst

```diff
@@ -0,0 +1,146 @@
+#. `See the guidance on Changelogs <https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs>`_ and include a `changelog fragment <https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment>`_ if appropriate.
+
+
+Running Tests Locally
+---------------------
+
+If you're making anything more than very small or one-time changes, you'll want to run the tests locally to avoid having to push a commit for each thing, and waiting for the CI to run tests.
+
+First, `review the guidance on testing collections <https://docs.ansible.com/ansible/devel/dev_guide/developing_collections_testing.html#testing-collections>`_, as it applies to this collection as well.
+
+Integration Tests
+-----------------
+
+Unlike other collections, we now require an `integration_config.yml <https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html#integration-config-yml>`_ file for properly running integration tests, as the tests require external dependencies (like a Vault server) and they need to know where to find those dependencies.
+
+If you have contributed to this collection or to the ``hashi_vault`` lookup plugin in the past, you might remember that the integration tests used to download, extract, and run a Vault server during the course of the tests, by default. This **legacy method** is not recommended but is still available (for now) via opt-in.
+
+Skip to the next section for a method that's nearly as easy but better off in the long run (docker-compose).
+
+Legacy Mode
+...........
+
+To get started quickly without having to set anything else, you can use legacy mode by copying the included integration config sample:
+
+.. code-block:: bash
+
+    $ cp tests/integration/integration_config.yml.sample tests/integration/integration_config.yml
+
+That file has everything configured to be able to run the integration tests and have them set up the dependencies for you.
+
+You will also need the following additional Ansible collections:
+
+* `community.crypto <https://galaxy.ansible.com/community/crypto>`_
+* `community.general <https://galaxy.ansible.com/community/general>`_ (MacOS local/venv only)
+
+Running legacy mode tests in docker (recommended):
+
+.. code-block:: bash
+
+    $ ansible-test integration --docker default -v
+
+Running legacy mode tests in a controlled python virtual environment (**not recommended**):
+
+.. code-block:: bash
+
+    $ ansible-test integration --venv --requirements --allow-destructive -v
+
+Note that your system packages may be manipulated by running locally or in a venv.
+
+Legacy mode is not recommended because a new Vault server and proxy server will be downloaded, set up, configured, and/or uninstalled, for every *target*. Traditionally, we've only had one target, and so it was a good way to go, but that's no longer going to be the case. This is going to make it slower and slower as you'll incur the overhead on every target, in every run.
```

Comment:
This is a really convincing argument. I'd move it up to the beginning of the section, so folks might read it before they follow the instructions on running tests in legacy mode.

Reply:
Perfecto 👌 moved into a nice note box, slightly re-worded, thank you!